src/index.js

```diff
@@ -75,7 +68,10 @@ export default function profileStore(options = {}) {
 			performance.clearMarks(markName + START)
 			performance.clearMarks(markName + END)
 			performance.clearMeasures(getMarkLabel(markInfo, markType))
-			counter -= counterDelta
+			counter[markInfo.name]--
+			if (counter[markInfo.name] <= 0) {
+				delete counter[markInfo.name]
+			}
```

Comment:
The more I think the more I like your version. Reverting to it to merge the MR. Sorry for confusion.
```suggestion
			if (counter[markInfo.name]) {
				counter[markInfo.name]--
			} else {
				delete counter[markInfo.name]
			}
```